### PR TITLE
new: forge command compat layer

### PR DESCRIPTION
**Pull request by Wyvest**
## Description
adds combat for forge commands in oneconfig

## Checklist
- [x] I made a clear description of what was changed
- [x] I stated why these changes were necessary
- [x] I updated documentation or said what needs to be updated
- [x] I made sure these changes are backwards compatible
- [x] This pull request is for one feature/bug fix

### CODE REVIEW

File: versions/src/main/java/cc/polyfrost/oneconfig/internal/OneConfig.java [hunk 1]
```diff
@@ -157,6 +146,53 @@ public static void init() {
...
+            }
+        }
+        for (net.minecraft.command.ICommand command :  net.minecraftforge.client.ClientCommandHandler.instance.getCommands().values()) {
+            if (Objects.equals(command.getCommandName(), mod.getModId())) {
+                ForgeCompat.compatMods.put(new ForgeCompat.ForgeCompatMod(mod.getName(), ModType.THIRD_PARTY), () -> new TickDelay(() -> {
+                    UScreen.displayScreen(null);
+                    try {
+                        command.processCommand(UPlayer.getPlayer(), new String[]{});
+                    } catch (Exception e) {
+                        UChat.chat(ChatColor.RED + "Forge command compat has failed! Please report this to Polyfrost on https://inv.wtf/polyfrost!");
```
Comment (xtrm-en) [inline versions/src/main/java/cc/polyfrost/oneconfig/internal/OneConfig.java:165]: maybe change it to https://polyfrost.cc/discord

**Review by xtrm-en — APPROVED** [review]

**Review by xtrm-en — DISMISSED** [review]
:+1:

**Review by xtrm-en — APPROVED** [review]
still good

**Review by xtrm-en — APPROVED** [review]
i swear to god wyvest

File: versions/src/main/java/cc/polyfrost/oneconfig/internal/OneConfig.java [hunk 2]
```diff
@@ -157,6 +146,58 @@ public static void init() {
...
+            }
+        }
+        for (net.minecraft.command.ICommand command :  net.minecraftforge.client.ClientCommandHandler.instance.getCommands().values()) {
+            if (Objects.equals(command.getCommandName(), mod.getModId())) {
+                ForgeCompat.compatMods.put(new ForgeCompat.ForgeCompatMod(mod.getName(), ModType.THIRD_PARTY), () -> new TickDelay(() -> {
+                    UScreen.displayScreen(null);
+                    try {
+                        //execute
+                        //#if MC<=10809
+                        command.processCommand(UPlayer.getPlayer(), new String[]{});
```
Comment (u9g) [inline versions/src/main/java/cc/polyfrost/oneconfig/internal/OneConfig.java:165]: Why not passing any args to the command?

**Review by u9g — COMMENTED** [review]
I looked over it and this is the only thing that sticks out to me as weird, assuming you tested it, it probably works, but would like a reply on the review.

File: versions/src/main/java/cc/polyfrost/oneconfig/internal/OneConfig.java (hunk 2 above)
Comment (Wyvest) [reply versions/src/main/java/cc/polyfrost/oneconfig/internal/OneConfig.java:165]: because usually that opens the config

**Review by nxtdaydelivery — APPROVED** [review]
yes

**Pull request merged**